src/main/java/com/github/hokkaydo/eplbot/module/code/PythonRunner.java

```diff
@@ -0,0 +1,83 @@
+package com.github.hokkaydo.eplbot.module.code;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import com.github.hokkaydo.eplbot.Strings;
+import java.io.File;
+import java.io.FileWriter;
+public class PythonRunner {
+    private static final String CURRENT_DIR = System.getProperty("user.dir") + "\\src\\temp\\";
+    public static String run(String input) {
+        if (containsUnsafeKeywords(input)){
+            return "Compilation failed:\nCheck if 'exec' or 'eval' are used";
+        }
+        try {
+            File sourceFile = new File(CURRENT_DIR,"temp.py");
+            sourceFile.deleteOnExit();
+            FileWriter writer = new FileWriter(sourceFile);
+            writer.write(input);
+            writer.close();
+            Process process = new ProcessBuilder("python",  sourceFile.getAbsolutePath()).redirectErrorStream(true).start();
+            Thread timeoutThread = new Thread(() -> {
+                try {
+                    Thread.sleep(1000 * Integer.parseInt(Strings.getString("COMMAND_CODE_TIMELIMIT")));
+                    process.destroy();
+                } catch (InterruptedException e) {
+                }
+            });
+            timeoutThread.start();
+            StringBuilder output = new StringBuilder();
+            String line;
+            while ((line = new BufferedReader(new InputStreamReader(process.getInputStream())).readLine()) != null) {
+                output.append(line).append("\n");
+            }
+            if (process.waitFor() == 0) {
+                timeoutThread.interrupt();
+                deleteFiles();
+                return output.toString();
+            } else {
+                timeoutThread.interrupt();
+                deleteFiles();
+                String outProcess = output.toString();
+                if (outProcess.isEmpty()) {
+                    return "Run failed: Timelimit exceeded " + Strings.getString("COMMAND_CODE_TIMELIMIT") + " s";
+                } else {
+                    return "Run failed:\n" + outProcess;
+                }
+            }
+        } catch (IOException | InterruptedException e) {
```

Comment:
```suggestion
        } catch (IOException e) {
```
already caught earlier

Reply:
int java.lang.Process.waitFor() throws InterruptedException
if (process.waitFor() == 0) might return InterruptedException while the first ignored one comes from the Thread

Reply:
I probably missed it